### ISSUE TITLE
Unable to get data in the gadgets on jira dashboard

### ISSUE THREAD

**Issue by varmagadiraju** [zenoss/testrail-jira-gadgets#15]
Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure script '<URL>'. This request has been blocked; the content must be served over HTTPS.

Mixed Content: The page at 'https://jira.qualifacts.org/secure/Dashboard.jspa?selectPageId=14208' was loaded over HTTPS, but requested an insecure script 'http://testrailgadget.qualifacts.org/gadgets/testrail-project-summary.js'. This request has been blocked; the content must be served over HTTPS.

**Comment by varmagadiraju** [comment]
I saw the same error for all gadgets.

**Comment by varmagadiraju** [comment]
Hello, Support team any update on my issue? it is a very high priority issue could you please check this, thanks.

**Comment by AwesomeDonYoung** [comment]
If your Jira server is HTTPS then the web server with the gadgets (the XML files) also has to be HTTPS. It's a requirement from the Jira server.

**Comment by varmagadiraju** [comment]
Thanks for replying Don Young,

We setup both(Jira and Testrail-Gadget) servers on HTTPS.

**Comment by varmagadiraju** [comment]
Jira Server: https://jira.qualifacts.org
Testrail-Gadget Server: http://testrailgadget.qualifacts.org

**Comment by AwesomeDonYoung** [comment]
The value for the gadgetURL user preference in one or all of the following files might be using http: instead https:

- gadgets/testrail-projects-all-summary.xml
- gadgets/testrail-release-summary.xml
- scripts/template-*.xml (6 files)

If the ones in the scripts directory are updated the Perl scripts will need to re-run to update the non-template versions. And you'll probably need to clear your browser cache to ensure the new XML is loaded on the dashboard page.

**Comment by varmagadiraju** [comment]
This we fix, now we have a new issue. I'm unable to edit some gadgets. Here is what I'm getting.
![image](https://user-images.githubusercontent.com/67124491/87199878-049b8880-c2c1-11ea-862a-cf62ec5865ba.png)

**Comment by varmagadiraju** [comment]
![image](https://user-images.githubusercontent.com/67124491/87199926-2268ed80-c2c1-11ea-9563-af2b6eb4eff7.png)

**Comment by varmagadiraju** [comment]
![image](https://user-images.githubusercontent.com/67124491/87315090-a69db980-c4e9-11ea-9835-d82a47022364.png)

This three gadgets I'm unable to edit, what is the issue? could you please help us. Thanks

**Comment by varmagadiraju** [comment]
Hello buddy,

Any update on my request?

**Comment by AwesomeDonYoung** [comment]
It is an issue I reproduced with Jira 8. It works fine with Jira 6 and 7. I never found any issue with the gadget code, and Atlassian closed the defect I opened on Jira as "unreproducible".

**Comment by varmagadiraju** [comment]
Don Young - Thanks for replying, we are using Jira 8.x, so that means these four gadgets we can't edit, am I correct

**Comment by AwesomeDonYoung** [comment]
Correct. For any gadget that has "settings" you will be unable to edit. If I remember correctly clicking Edit doesn't do anything. The one gadget that will work is "TestRail: Testing Status" because it doesn't have any settings to change. It displays the current status for all test plans and test runs in all projects.

**Comment by varmagadiraju** [comment]
I found 4 gadgets that are not working.
- TestRail: Daily Activity Summary
- TestRail: Milestone Status
- TestRail: Release Status
- TestRail: User Activity Summary

**Comment by varmagadiraju** [comment]
Hello Don Young,

I'm unable to edit any gadgets, by default it is taking some data I don't know? Could you please help us. Thanks

**Comment by varmagadiraju** [comment]
Also, Our Jira version is 8.5.1

**Comment by AwesomeDonYoung** [comment]
On Jira 8.x the Edit button doesn't work. As near as I can tell it is an issue with Jira, but Atlassian couldn't reproduce it. The only gadget that should work on Jira 8 is "TestRail: Testing Status".

**Comment by varmagadiraju** [comment]
Even that one also I'm not able to edit.

**Comment by varmagadiraju** [comment]
In the time of setting, I saw some are editable but now that option is not available.

**Comment by varmagadiraju** [comment]
![image](https://user-images.githubusercontent.com/67124491/87450867-b50cd380-c5c4-11ea-962b-16eb08b428fe.png)
 if you see the image by default it is picking one option.

**Comment by AwesomeDonYoung** [comment]
The script(s) that generate the xml files get the list of projects and items (i.e. plans, runs, etc.) from TestRail and put it in that xml file. The first one that gets returned is the what gets set as the default. You can manually edit that XML file and set one of the other values as the default, but you'll need to make sure the cron job to run the scripts is disabled.

And "TestRail: Testing Status" doesn't have an edit button. It gets all of the plans/runs in all of the projects and displays their current status.

**Comment by gbaileyxevo** [comment]
FWIW I have the gadgets added to our confluence instance and those you can edit and choose runs etc.
The list of runs in the gadget configuration looks to be just the basic strings from the XML, not parsed in anyway so tough to read and find the run you are looking for.  But at least using the gadgets in confluence allows you edit them and choose what you need, where in Jira you can't.

btw we are running Confluence Server 7.5.2.  Our jira instance is running 8.5.3
